boot/freeldr/freeldr/include/mm.h

```diff
@@ -51,7 +51,12 @@ typedef struct _FREELDR_MEMORY_DESCRIPTOR
 #define MM_PAGE_SIZE    4096
 #define MM_PAGE_MASK    0xFFF
 #define MM_PAGE_SHIFT    12
-#define MM_MAX_PAGE        0xFFFFF
+#if defined(_X86PAE_)
```

Comment:
Just for the record: I don't think that we will actually compile freeloader with that definition, and I don't think freeloader should bother with this, but PAE is not really a concern for me other than as a design principle, so it basically serves as a documentation, how stuff should work, and an actual correct implementation would probably need to either make sure Mm can deal with too high physical pages or freeloader needs to determine the kernel that is loaded and decide the margin based upon that.

Reply:
Especially since it is known, that the same NTLDR on Windows can manage both the non-PAE and the PAE ntoskrnl.